ci/start_init_test.sh

```diff
@@ -834,13 +831,49 @@ run_data_integrity_test() {
 	else
 		echo "Unable to detect iSCSI device, login failed"; collect_logs_and_exit
 	fi
-	#Cleanup is not being performed here because this data will be used to test clone feature in the next test
 }
 
-create_snapshot() {
-	echo "--------------create_snapshot-------------"
-	id=`curl http://$1:9501/v1/volumes | jq '.data[0].id' |  tr -d '"'`
-	curl -H "Content-Type: application/json" -X POST -d '{"name":"snap1"}' http://$CONTROLLER_IP:9501/v1/volumes/$id?action=snapshot
+run_data_integrity_test() {
+	echo "--------------------Run_data_integrity_test------------------"
+	orig_controller_id=$(start_controller "$CONTROLLER_IP" "store1" "3")
+	replica1_id=$(start_replica "$CONTROLLER_IP" "$REPLICA_IP1" "vol1")
+	replica2_id=$(start_replica "$CONTROLLER_IP" "$REPLICA_IP2" "vol2")
+	replica3_id=$(start_replica "$CONTROLLER_IP" "$REPLICA_IP3" "vol3")
+	sleep 5
+	test_data_integrity
+	#Cleanup is not being performed here because this data will be used
+	#to test snapshot feature in the next test.
+}
+
+# create_and_verify_snapshot creates a snapshot and verifies if it
+# has message given below (success case) or not (failure case).
+# As a part of negative test we are trying to create the snapshot with the
+# same name which returns snapshot_exists.
+create_and_verify_snapshot() {
+	snap='"snap'$2'"'
+	# remove trailing double quotes
+	snapname=$(echo $snap | tr -d '"')
+	message=`curl -H "Content-Type: application/json" -X POST -d '{"name":"snap'$2'"}' http://$CONTROLLER_IP:9501/v1/volumes/$1?action=snapshot | jq '.message' | tr -d '"'`
+	if [ "$message" == "Snapshot: $snapname created successfully" ] || [ "$message" == "Snapshot: $snapname already exists" ] ;
```

Comment:
In both successful and failure cases we are returning success.

Reply:
Thanks for pointing out this. Will update it.